tests/Unit/RequestPaginationTest.php

```diff
@@ -0,0 +1,189 @@
+        });
+
+    expect($responses)->toHaveCount(4)->each->toBeInstanceOf(Response::class)
+        ->and($superheroes)->toHaveCount(20)->each->toBeArray();
+});
+
+test('you can continue a paginator from where it left off', function (): void {
+    $connector = new PageRequestPaginatorConnector;
+    $request = new PageGetSuperHeroesRequest;
+
+    $paginator = $connector->paginate($request);
+
+    $responses = [];
+    $superheroes = [];
+
+    foreach ($paginator as $response) {
+        $responses[] = $response;
+        $superheroes = [...$superheroes, ...$response->json('data')];
+
+        break;
+    }
+
+    expect($responses)->toHaveCount(1)->each->toBeInstanceOf(Response::class)
+        ->and($superheroes)->toHaveCount(5)->each->toBeArray();
+
+    // TODO: See the test skip() message.
+
+    foreach ($paginator as $response) {
+        $responses[] = $response;
+        $superheroes = [...$superheroes, ...$response->json('data')];
+    }
+
+    expect($responses)->toHaveCount(4)->each->toBeInstanceOf(Response::class)
+        ->and($superheroes)->toHaveCount(20)->each->toBeArray();
+})->skip(<<<'REASON'
+    Because we're breaking out of the previous loop, `Iterator::next()` won't be executed.
+    Which means that we'll redo the same request. We need to find a way to properly continue where we left off.
+REASON);
+
+test('you can iterate a paginator in for loops', function (): void {
+    $connector = new PageRequestPaginatorConnector;
+    $request = new PageGetSuperHeroesRequest;
+
+    $paginator = $connector->paginate($request);
+
+    $responses = [];
+    $superheroes = [];
+
+    for (; $paginator->valid(); $paginator->next()) {
+        $responses[] = $response = $paginator->current();
+        $superheroes = [...$superheroes, ...$response->json('data')];
+    }
+
+    expect($responses)->toHaveCount(4)->each->toBeInstanceOf(Response::class)
+        ->and($superheroes)->toHaveCount(20)->each->toBeArray();
+});
+
+test('you can automagically rewind a paginator, starting over from the start in a new loop', function (): void {
+    $connector = new PageRequestPaginatorConnector;
+    $request = new PageGetSuperHeroesRequest;
+
+    $paginator = $connector->paginate($request);
+    $paginator->enableRewinding();
+
+    $responses = [];
+    $superheroes = [];
+
+    foreach ($paginator as $response) {
+        $responses[] = $response;
+        $superheroes = [...$superheroes, ...$response->json('data')];
+
+        break;
+    }
+
+    expect($responses)->toHaveCount(1)->each->toBeInstanceOf(Response::class)
+        ->and($superheroes)->toHaveCount(5)->each->toBeArray();
+
+    foreach ($paginator as $response) {
+        $responses[] = $response;
+        $superheroes = [...$superheroes, ...$response->json('data')];
+    }
+
+    expect($responses)->toHaveCount(5)->each->toBeInstanceOf(Response::class)
+        ->and($superheroes)->toHaveCount(25)->each->toBeArray();
```

Comment:
Again I would have a pest function that is a shortcut to get the full array of superheroes in the right order so you know exactly the API has returned with what you expect